shared/chat/conversation/messages/wrapper/exploding-height-retainer/index.desktop.js

```diff
@@ -4,40 +4,73 @@ import * as ReactDOM from 'react-dom'
 import {resolveRootAsURL} from '../../../../../desktop/app/resolve-root.desktop'
 import {urlsToImgSet} from '../../../../../common-adapters/icon.desktop'
 import {Box, ConnectedUsernames, Text} from '../../../../../common-adapters'
-import {collapseStyles, globalColors, styleSheetCreate} from '../../../../../styles'
+import {
+  collapseStyles,
+  glamorous,
+  globalColors,
+  platformStyles,
+  styleSheetCreate,
+} from '../../../../../styles'
 import type {Props} from '.'
 
 const explodedIllustration = resolveRootAsURL('../images/icons/pattern-ashes-desktop-400-68.png')
 const explodedIllustrationUrl = urlsToImgSet({'68': explodedIllustration}, 68)
 
-const messageHeights = {}
+const copyChildren = children =>
+  React.Children.map(children, child => (child ? React.cloneElement(child) : child))
+
+const animationDuration = 1.5
+
+const retainedHeights = {}
 
 type State = {
-  height: ?number,
+  animating: boolean,
+  children: ?React.Node,
+  height: number,
 }
 class ExplodingHeightRetainer extends React.Component<Props, State> {
-  state = {height: 17}
+  state = {animating: false, children: copyChildren(this.props.children), height: 17}
+  timeoutID: ?TimeoutID = null
 
-  constructor(props: Props) {
-    super(props)
-    if (messageHeights[props.messageKey]) {
-      this.state = {height: messageHeights[props.messageKey]}
+  static getDerivedStateFromProps(nextProps: Props, prevState: State) {
+    return nextProps.retainHeight ? null : {children: copyChildren(nextProps.children)}
+  }
+
+  componentDidMount() {
+    // remeasure if we are already exploded
+    if (this.props.retainHeight && retainedHeights[this.props.messageKey] && this.props.measure) {
+      delete retainedHeights[this.props.messageKey]
+      this.props.measure()
     }
   }
 
-  componentDidUpdate() {
+  componentDidUpdate(prevProps: Props) {
     if (this.props.retainHeight) {
+      if (!prevProps.retainHeight && !this.timeoutID) {
+        // destroy local copy of children when animation finishes
+        this.setState({animating: true}, () => {
+          this.timeoutID = setTimeout(
```

Comment:
likely need to clear the timeout if this gets his if one exists still

Reply:
done :+1: thanks